subprojects/maven/src/main/java/org/gradle/api/publication/maven/internal/action/DefaultMavenDeployRetrier.java

```diff
@@ -0,0 +1,90 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.publication.maven.internal.action;
+
+import org.apache.http.conn.HttpHostConnectException;
+import org.apache.maven.artifact.ant.RemoteRepository;
+import org.gradle.internal.UncheckedException;
+import org.gradle.internal.resource.transport.http.HttpErrorStatusCodeException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.sonatype.aether.deployment.DeploymentException;
+
+import java.net.SocketTimeoutException;
+import java.util.concurrent.Callable;
+
+public class DefaultMavenDeployRetrier implements MavenDeployRetrier {
+    private static final Logger LOGGER = LoggerFactory.getLogger(DefaultMavenDeployRetrier.class);
+    private final RemoteRepository remoteRepository;
+    private final Callable operation;
+    private final int maxDeployAttempts;
+    private final int initialBackOff;
+
+    public DefaultMavenDeployRetrier(Callable operation, RemoteRepository remoteRepository, int maxDeployAttempts, int initialBackOff) {
+        this.operation = operation;
+        this.maxDeployAttempts = maxDeployAttempts;
+        this.initialBackOff = initialBackOff;
+        assert maxDeployAttempts > 0;
+        assert initialBackOff > 0;
+        this.remoteRepository = remoteRepository;
+    }
+
+
+    public void deployWithRetry() throws DeploymentException {
+        int backoff = initialBackOff;
+        int retries = 0;
+        while (retries < maxDeployAttempts) {
+            retries++;
+            Throwable failure;
+            try {
+                operation.call();
+                if (retries > 1) {
+                    LOGGER.info("Successfully deployed resource after {} retries", retries - 1);
+                }
+                break;
+            } catch (Exception throwable) {
+                failure = throwable;
+            }
+            if (!isLikelyTransientNetworkingIssue(failure) || retries == maxDeployAttempts) {
+                throw new DeploymentException("Could not deploy to remote repository | " + failure.getMessage(), failure);
+            } else {
+                LOGGER.info("Error while accessing remote repository {}. Waiting {}ms before next retry. {} retries left", remoteRepository, backoff, maxDeployAttempts - retries, failure);
+                try {
+                    Thread.sleep(backoff);
+                    backoff *= 2;
+                } catch (InterruptedException e) {
+                    throw UncheckedException.throwAsUncheckedException(e);
+                }
+            }
+        }
+    }
+
+    private static boolean isLikelyTransientNetworkingIssue(Throwable failure) {
```

Comment:
I would look into extracting the similar logic found in `ErrorHandlingModuleComponentRepositoryAccess#isLikelyTransientNetworkingIssue` so that it can be re-used here instead of copied.

Reply:
This is now introduced in https://github.com/gradle/gradle/pull/9163/commits/81c7069e8ae96f07366e7b082387b7b8bc40b6ee 

Please let me know your thoughts